src/components/PageMain.tsx

```diff
@@ -0,0 +1,64 @@
+import SearchPanel from './SearchPanel';
+import SearchResult from './SearchResult';
+import React from 'react';
+import { IState, IItem } from '../types';
+import Loader from './Loader';
+
+interface I {}
+
+export default class PageMain extends React.Component {
+  state: IState;
+  constructor(props: I | Readonly<I>) {
+    super(props);
+    this.state = {
+      searchQuery: localStorage.getItem('searchQuery') ?? '',
+      items: [],
+      isLoading: true,
+    };
+  }
+
+  async componentDidMount() {
+    const { searchQuery } = this.state;
+    await this.fetchData(searchQuery || '');
+  }
+
+  async fetchData(query: string) {
+    this.setState({ items: [], isLoading: true });
+    fetch('https://stapi.co/api/v2/rest/book/search')
+      .then((res) => res.json())
+      .then((json) => {
+        const select = query
+          ? json.books.filter((el: IItem) => el.title.includes(query))
+          : json.books;
+        this.setState({ items: select });
+        this.setState({ isLoading: false });
+      });
+  }
```

Comment:
better to move this into separated module. You can call it API or whatever you like